pkg/kubelet/container/helpers_test.go

```diff
@@ -459,7 +459,7 @@ func TestShouldContainerBeRestarted(t *testing.T) {
 		for i, policy := range policies {
 			pod.Spec.RestartPolicy = policy
 			e := expected[c.Name][i]
-			r := ShouldContainerBeRestarted(&c, pod, podStatus)
+			r := ShouldContainerBeRestarted(pod, podStatus.FindContainerStatusByName(c.Name), false)
```

Comment:
Overall looks good to me. However, `FindContainerStatusByName` will loop all container status.

In original logic, when `pod.DeletionTimestamp != nil` is true, it can skip FindContainerStatusByName.

I wondered if the benefit of  this pr outweighs the small performance penalty. Not sure.

Reply:
Thanks review, I don't think there should be any improvement in performance，because the ```pod.DeletionTimestamp == nil```,  the original logic will also execute this method twice，but now only execute once.